lib/puppet/node/environment.rb

```diff
@@ -82,8 +82,12 @@ def self.remote(name)
   # @param name [Symbol] The environment name
   def initialize(name, modulepath, manifest, config_version)
     @name = name
-    @modulepath = modulepath
-    @manifest = manifest
+    @modulepath = modulepath.map { |p| Puppet::FileSystem.expanded_path_string(p) }
+    if manifest && !manifest.is_a?(Symbol) && Puppet::FileSystem.exist?(manifest)
```

Comment:
I can debug and let you know for sure the values.. but while debugging calls that entered here, I saw several different symbols come through as the value of `manifest`, in addition to actual path strings.  I wasn't certain what they meant - and also wasn't sure if we got both `Pathname` and `String` instances here - so that's why I didn't call `exist?` on only string values.

Will report back on this one.